x/programs/rust/examples/automated-market-maker/src/lib.rs

```diff
@@ -19,16 +19,24 @@ const MAX_GAS: Gas = 10000000;
 
 /// Initializes the pool with the two tokens and the liquidity token
 #[public]
-pub fn init(context: &mut Context, token_x: Program, token_y: Program, liquidity_token: Program) {
+pub fn init(context: &mut Context, token_x: Program, token_y: Program, liquidity_token: Id) {
+    let lt_program = context.deploy(liquidity_token, &[0, 1]);
+    let liquidity_context = ExternalCallContext::new(lt_program, MAX_GAS, 0);
+    token::init(
+        &liquidity_context,
+        String::from("liquidity token"),
+        String::from("LT"),
+    );
+
     context
         .store((
             (TokenX, token_x),
             (TokenY, token_y),
-            (LiquidityToken, liquidity_token),
+            (LiquidityToken, lt_program),
         ))
         .expect("failed to set state");
```

Comment:
this should be last line, probably a weird rebase

Reply:
done